lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""In this class, we manage backup configurations and actions.
+
+Specifically backups are handled with Percona Backup MongoDB (pbm) which is installed as a snap
+during the install phase. A user for PBM is created when MongoDB is first started during the
+start phase.
+"""
+import logging
+
+from charms.mongodb.v0.helpers import generate_password
+from charms.mongodb.v0.mongodb import MongoDBConfiguration
+from charms.operator_libs_linux.v1 import snap
+from ops.framework import Object
+from ops.model import ActiveStatus
+
+# The unique Charmhub library identifier, never change it
+LIBID = "18c461132b824ace91af0d7abe85f40e"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+
+
+PBM_S3_CONFIGS = [
+    ("storage.s3.region", "s3-storage-region"),
+    ("storage.s3.bucket", "s3-storage-bucket"),
+    ("storage.s3.prefix", "s3-storage-prefix"),
+    ("storage.s3.credentials.access-key-id", "s3-access-key-id"),
+    ("storage.s3.credentials.secret-access-key", "s3-secret-access-key"),
+    ("storage.s3.serverSideEncryption.kmsKeyID", "s3-kms-key-id"),
+]
+
+
+class MongoDBBackups(Object):
+    """In this class, we manage mongodb backups."""
+
+    def __init__(self, charm, substrate="k8s"):
+        """Manager of MongoDB client relations."""
+        super().__init__(charm, "client-relations")
+        self.charm = charm
+        self.substrate = substrate
+        self.framework.observe(self.charm.on.config_changed, self._on_pbm_config_changed)
+
+    def _on_pbm_config_changed(self, event) -> None:
```

Comment:
Why do we install snap on config_changed? I think we have a special event for such actions "on install"

Reply:
We don't this only checks if it has been installed. (we cannot configure pbm snap if it hasn't been installed)